superbench/benchmarks/micro_benchmarks/ort_inference_performance.py

```diff
@@ -0,0 +1,187 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""TensorRT inference micro-benchmark."""
+
+import time
+import statistics
+from pathlib import Path
+
+import torch
+import torchvision.models
+import numpy as np
+import onnxruntime as ort
+from onnxruntime.quantization import quantize_dynamic
+
+from superbench.common.utils import logger
+from superbench.benchmarks import BenchmarkRegistry, Platform, Precision
+from superbench.benchmarks.micro_benchmarks import MicroBenchmark
+
+
+class ORTInferenceBenchmark(MicroBenchmark):
+    """ONNXRuntime inference micro-benchmark class."""
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+
+        self._pytorch_models = [
+            'resnet50',
+            'resnet101',
+            'resnet152',
+            'densenet169',
+            'densenet201',
+            'vgg11',
+            'vgg13',
+            'vgg16',
+            'vgg19',
+        ]
+        self.__graph_opt_level = {
+            0: ort.GraphOptimizationLevel.ORT_DISABLE_ALL,
+            1: ort.GraphOptimizationLevel.ORT_ENABLE_BASIC,
+            2: ort.GraphOptimizationLevel.ORT_ENABLE_EXTENDED,
+            3: ort.GraphOptimizationLevel.ORT_ENABLE_ALL,
+        }
+        self.__model_cache_path = Path(torch.hub.get_dir()) / 'checkpoints'
+
+    def add_parser_arguments(self):
+        """Add the specified arguments."""
+        super().add_parser_arguments()
+
+        self._parser.add_argument(
+            '--pytorch_models',
+            type=str,
+            nargs='+',
+            default=self._pytorch_models,
+            help='ONNX models for TensorRT inference benchmark, e.g., {}.'.format(', '.join(self._pytorch_models)),
+        )
+
+        self._parser.add_argument(
+            '--precision',
+            type=Precision,
+            choices=[Precision.FLOAT32, Precision.FLOAT16, Precision.INT8],
+            default=Precision.FLOAT16,
+            required=False,
+            help='Precision for inference, allow int8, float16, or float32 only.',
+        )
+
+        self._parser.add_argument(
+            '--graph_opt_level',
+            type=int,
+            default=3,
+            choices=[0, 1, 2, 3],
+            required=False,
+            help='ONNXRuntime graph optimization level, 0 for ORT_DISABLE_ALL, 1 for ORT_ENABLE_BASIC, '
+            '2 for ORT_ENABLE_EXTENDED, 3 for ORT_ENABLE_ALL.',
+        )
+
+        self._parser.add_argument(
+            '--batch_size',
+            type=int,
+            default=32,
```

Comment:
Do we need to change the default batch size to 1? It's also for Yifan's TensorRT PR

Reply:
For all the default values of benchmarks, we used the most common values, not the least value.

Reply:
Do you think 32 is reasonable for inference? It's my question here.

Reply:
There is a balance issue among latency/throughput/resource utilization when using batchsize as 1 and greater than 1. For inference by using GPU, I think batch size greater than 1 is reasonable, since GPU resource is not shared with other services, batch-size as 1 is very wasteful and I think no one will use in this way. For inference by CPU, we can set it as 1 to get the smallest latency.
Here we use the benchmarks for GPU validation, so we can set the default value as 32 like model training benchmarks. Do you think it is reasonable?

Reply:
Shall we check it on MLPerf? Whether it's a large number?

Reply:
Yifan said that Nvidia uses 32.